feat(tutorial): update tutorial to use the identity provider in the identity bundle (#89)
canonical/jaas-documentation: tutorial/index.md

```text

> # enable necessary add-ons

Reviewer: In general, I feel like there are too many of these explanations inside the code blocks. It doesn't read as well when you put the explanations in the code blocks. I'd opt to put the explanations outside the code blocks where possible and leave the code blocks as just code (in most cases).
So here, 
>Make sure MicroK8s is correctly set up by enabling the following addons and configure metallb.
> (code block)

Reviewer: A lot of these also already exist, so if you feel like it would be too many unnecessary changes then we can revisit it later.

Author: yeah, i've amended mine. But i agree with you we should get rid of them, it is clearer to read and c/p commands

> sudo microk8s enable dns host-access

Reviewer 2: Is ingress no longer necessary?

Author: nope, because we use traefik which works on metallb

Running `juju status` should now two offers that we will use from a different model in the next step.

### Create an user.
```text

> # disable MFA to avoid unnecessary steps

Reviewer: I think this is fine in the code block.

juju config kratos enforce_mfa=False
# create the user and get the identity-id
juju run kratos/0 create-admin-account email=test@example.com password=test username=admin
# reset the password to make it active
juju add-secret password-secret password=abc
juju grant-secret password-secret kratos
juju run kratos/0 reset-password identity-id=<identity-id> password-secret-id=<secret:id>
```

> ### Expose the identity bundle to your host machine.

Reviewer: This should have a note  below that it's only necessary if running in Multipass.

The reason you need to expose the identity bundle to your host machine is that at the end of this tutorial
you will need to login via a web browser.

```text

> # locate the ip of your multipass instance. It should be the ip for second interface.

Reviewer: The language should be more professional here, and if possible, more certain.
Since we can get it with `multipass list` as you mentioned just say that. Some like
```suggestion
# locate the ip of your multipass instance by running `multipass list`
```

# you can double check it by doing `multipass list` in your host machine.

> ip addr

Reviewer: Why this rather than `multipass list`?

Author: Because multipass list gives you multiple ip, and just one is the right one to make the routing works. And it is the one from `ip addr`

Author: I am not sure if it would always be the first one listed in `multipass list`

Reviewer: I don't really understand. On my system, 
```
$ multipass list
Name                    State             IPv4             Image
decisive-chiffchaff     Running           10.225.162.46    Ubuntu 22.04 LTS
snapcraft-jaas          Running           10.225.162.207   Ubuntu 20.04 LTS
```
and
```
$ ip addr
1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN group default qlen 1000
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
    inet 127.0.0.1/8 scope host lo
       valid_lft forever preferred_lft forever
    inet6 ::1/128 scope host noprefixroute 
       valid_lft forever preferred_lft forever
2: wlp2s0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP group default qlen 1000
    link/ether 48:e7:da:5b:b5:9b brd ff:ff:ff:ff:ff:ff
    inet 10.0.0.109/24 brd 10.0.0.255 scope global dynamic noprefixroute wlp2s0
       valid_lft 56522sec preferred_lft 56522sec
    inet6 fe80::f877:1559:f6d2:4ed8/64 scope link noprefixroute 
       valid_lft forever preferred_lft forever
3: mpqemubr0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP group default qlen 1000
    link/ether 52:54:00:0b:ab:27 brd ff:ff:ff:ff:ff:ff
    inet 10.225.162.1/24 brd 10.225.162.255 scope global mpqemubr0
       valid_lft forever preferred_lft forever
    inet6 fe80::5054:ff:fe0b:ab27/64 scope link 
       valid_lft forever preferred_lft forever
4: tap-6a244e8b928: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc fq_codel master mpqemubr0 state UP group default qlen 1000
    link/ether 66:47:93:8c:db:a8 brd ff:ff:ff:ff:ff:ff
    inet6 fe80::6447:93ff:fe8c:dba8/64 scope link 
       valid_lft forever preferred_lft forever
5: lxdbr0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP group default qlen 1000
    link/ether 00:16:3e:60:af:1d brd ff:ff:ff:ff:ff:ff
    inet 10.16.149.1/24 scope global lxdbr0
       valid_lft forever preferred_lft forever
7: veth5cdc8297@if6: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue master lxdbr0 state UP group default qlen 1000
    link/ether fe:aa:c0:96:37:ed brd ff:ff:ff:ff:ff:ff link-netnsid 0
..... <it goes on>
```
How do we know the address of the 2nd interface is always the correct one?
And where would you see multiple addresses from `multipass list`?

Author: on my machine:
```console
[4:46:49] ➜  ~ multipass list
Name                    State             IPv4             Image
my-juju-vm              Running           10.42.203.241    Ubuntu 24.04 LTS
                                          10.0.1.1
                                          10.219.89.1
                                          10.1.32.128

```

Author: it's interesting that we have completely different output for both commands. Maybe the best bet is `first ip listed when you do multipass list`

Reviewer: Interesting, I think I'd also opt for the output from `multipass list`

juju config traefik-public external_hostname=<multipass-ip>
# port-forward traefik pod to expose the service outside of the multipass instance
sudo microk8s.kubectl port-forward traefik-public-0 443:443 --namespace=iam --address=<multipass_ip> &
```

To test you've managed to expose the service you can
```text

> # in your host machine
> curl -k https://<multipass-ip>/iam-hydra/health/ready
> # the response should be {"status":"ok"}

Reviewer: This is another example where I think it's confusing, the sentence above runs on into the code block. Just
```suggestion
Run the following on your host machine to test that you've successfully exposed:
$ curl -k https://<multipass-ip>/iam-hydra/health/ready
{"status":"ok"}
```
I'm not quite sure about the `$` but hopefully you get my point.

# Changes to the UUID value after deployment will likely result in broken permissions.
# Use a randomly generated UUID.
juju config jimm uuid=3f4d142b-732e-4e99-80e7-5899b7e67e59
# The address to reach JIMM, this will configure ingress and is also used for OAuth flows/redirects.

> juju config jimm dns-name=test-jimm.localhost

Reviewer: Don't we need this?

Author: i've taken a look at the code, and it is used to configure the nginx ingress. so it's not needed when we use traefik

Author: i can confirm it is unset in my multipass and it works.
So it's not needed.

```
  dns-name: 
    description: DNS hostname that JIMM is being served from.
    source: unset
    type: string
```

Reviewer: I double checked and it is needed, see `JIMM_DNS_NAME` but Traefik provides a default DNS name to jimm of you don't use a custom one so that's probably why it works. But please confirm the URL of jimm is what you expect.

# Changes to the UUID value after deployment will likely result in broken permissions.
# Use a randomly generated UUID.
juju config jimm uuid=3f4d142b-732e-4e99-80e7-5899b7e67e59
```

```

> # Look for the ip in the ingress app "Serving at <ip>"

Reviewer: Again here.

> # Look for the ip in for the ingress app "Serving at <ip>"

Reviewer 2: "ip in for" = ? Feels like it should be just "ip for the"

Author: ah yeah
